datanode/sqlstore/accounts.go

```diff
@@ -66,6 +68,7 @@ func (as *Accounts) checkCache(
 func (as *Accounts) updateCache(a entities.Account) {
 	key := accountCacheKey{a.PartyID, a.AssetID, a.MarketID, a.Type}
 	as.cache[key] = a
+	as.idToAccountCache[deterministicAccountID(&a)] = a
```

Comment:
Bit of a shame to have two caches for the same data. Is there a reason why `Obtain` can't also use one with the `deterministicAccountID()` key too?

Reply:
you'd effectivley have to add a mapping index, which given the size of the account object and number of them doesn't carry any material advantage but does add a bit of complexity (IMO at least ;-) )

Reply:
> Bit of a shame to have two caches for the same data. Is there a reason why `Obtain` can't also use one with the `deterministicAccountID()` key too?

actually, theres not much in it, I'll update to map the cache key

Reply:
> > Bit of a shame to have two caches for the same data. Is there a reason why `Obtain` can't also use one with the `deterministicAccountID()` key too?
> 
> actually, theres not much in it, I'll update to map the cache key

ok, bit more of a refactor than originally intending, but the code is better for it I think, the Obtain method is a bit of an odd one in that it doesn't return an account but rather updates the instance passed in, but there were (and afaik still are) good reasons for it being done that way.  @Phil might want to double check the refactor in case missed something dumb.